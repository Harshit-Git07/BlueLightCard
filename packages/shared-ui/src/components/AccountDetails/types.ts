export type AccountDetailsProps = {
  accountNumber: string;
  firstName: string;
  lastName: string;
};
