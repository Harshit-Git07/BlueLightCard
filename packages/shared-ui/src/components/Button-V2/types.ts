import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ButtonProps = PropsWithChildren & {
  className?: string;
  type?: 'button' | 'submit';
  disabled?: boolean;
  slim?: boolean;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  newTab?: boolean;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
  size?: ButtonSize;
  'data-testid'?: string;
};

export type ButtonSize = 'XSmall' | 'Small' | 'Large';
