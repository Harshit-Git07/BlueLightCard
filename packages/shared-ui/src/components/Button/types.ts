import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ColorToken = {
  bg: string;
  hover: string;
  focus: string;
  text: string;
  border: string;
};

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
};
