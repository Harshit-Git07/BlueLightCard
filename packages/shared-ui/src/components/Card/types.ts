import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { RefObject } from 'react';

export type CardState = 'default' | 'hover' | 'selected';

export interface CardProps {
  initialCardState?: CardState;
  cardTitle?: string;
  description?: string;
  truncateDescription?: boolean
  buttonTitle?: string;
  imageAlt?: string;
  imageSrc?: string;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  showImage?: boolean;
  showDescription?: boolean;
  showButton?: boolean;
  onClick?: () => void;
  ariaLabel?: string;
  className?: string;
  'data-testid'?: string;
  canHover?: boolean;
}

export interface CardContentProps {
  title: string;
  description?: string;
  showDescription?: boolean;
  centerContent?: boolean;
  ariaLabel?: string;
  descriptionRef: RefObject<HTMLParagraphElement>;
  truncateDescription?: boolean;
}
