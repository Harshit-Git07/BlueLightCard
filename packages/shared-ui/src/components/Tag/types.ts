import { PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';

export enum TagState {
  Default = 'Default',
  Success = 'Success',
  Warning = 'Warning',
  Error = 'Error',
  Info = 'Info',
}

export type TagProps = PropsWithChildren & {
  state?: 'Default' | 'Success' | 'Warning' | 'Error' | 'Info';
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  infoMessage?: string;
};
