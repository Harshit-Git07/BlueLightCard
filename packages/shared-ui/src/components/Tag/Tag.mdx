import { Meta, Canvas, Controls } from '@storybook/blocks';

import * as Tag from './Tag.stories';

<Meta of={Tag} />

# Tags

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [States](#states)
- [Design](#design)

## Overview

Tags are static components that show helpful details or attributes .Tags can be added to a card to display an attribute, quality or state of a particular item. They should be used sparingly to avoid overwhelming the user. They come in five styles for different use cases.

## Props

<Controls />

## Usage
➡️ Tags must appear near the item they're adding extra detail to. Any tags styles can be used together but are stronger when the same style is used.

➡️ Use for categorizing,status, labeling, or read-only situations

➡️ Do not use tags as links that direct you to an entirely different page or launch you from a current experience to a separate tab

➡️ Tag titles should be concise and informative

➡️ Tag titles should describe the tag in a few words or be under 20 characters when possible

➡️ Only include long title content in tags when necessary

## States

List of states can be shown below here.

### Default

<Canvas of={Tag.TagDefault} />

### Success

<Canvas of={Tag.TagSuccess} />

### Warning

<Canvas of={Tag.TagWarning} />

### Error

<Canvas of={Tag.TagError} />

### Info

<Canvas of={Tag.TagInfo} />

## Design
Please see the [Figma design](<https://www.figma.com/design/n2NzB4G3OTR74o3dQ4BOwm/My-Account-%26-Sign-up-Component-library?node-id=471-37795&node-type=instance&m=devq>) for the design of the Tag component.
