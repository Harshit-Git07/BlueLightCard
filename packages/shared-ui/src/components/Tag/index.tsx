import React, { FC } from 'react';
import { TagProps } from './types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { cssUtil } from '../../utils/cssUtil';
const Tag: FC<TagProps> = ({ state = 'Default', iconLeft, iconRight, infoMessage }) => {
  const colorTokens: Record<string, any> = {
    Default: {
      bg: 'bg-colour-surface-container-light dark:bg-colour-surface-container-dark',
      text: 'text-colour-onSurface-light dark:text-colour-onSurface-dark',
      border: 'border-[0.5px] border-colour-onSurface-light dark:border-colour-onSurface-dark',
    },
    Success: {
      bg: 'bg-colour-success-bright-light dark:bg-colour-success-bright-dark',
      text: 'text-colour-onSuccess-bright-light dark:text-colour-onSuccess-bright-dark',
      border:
        'border-[0.5px] border-colour-onSuccess-bright-light dark:border-colour-onSuccess-bright-dark',
    },
    Warning: {
      bg: 'bg-colour-warning-bright-light dark:bg-colour-warning-bright-dark',
      text: 'text-colour-onWarning-bright-light dark:text-colour-onWarning-bright-dark',
      border:
        'border-[0.5px] border-colour-onWarning-bright-light dark:border-colour-onWarning-bright-dark',
    },
    Error: {
      bg: 'bg-colour-error-bright-light dark:bg-colour-error-bright-dark',
      text: 'text-colour-onError-bright-light dark:text-colour-onError-bright-dark',
      border:
        'border-[0.5px] border-colour-onError-bright-light dark:border-color-onError-bright-dark',
    },
    Info: {
      bg: 'bg-colour-info-bright-light dark:bg-colour-info-bright-dark',
      text: 'text-colour-onInfo-bright-light dark:text-colour-onInfo-bright-dark',
      border:
        'border-[0.5px] border-colour-onInfo-bright-light dark:border-colour-onInfo-bright-dark',
    },
  };

  const tokenClasses = colorTokens[state] || colorTokens['Default'];
  const borderRadius = 'rounded';

  const classes = cssUtil([
    'inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1',
    tokenClasses.bg,
    tokenClasses.text,
    tokenClasses.border,
    borderRadius,
  ]);

  return (
    <div className={classes}>
      {iconLeft && (
        <span className="flex items-center">
          <FontAwesomeIcon className="mr-1 text-current" icon={iconLeft} size="sm" />
        </span>
      )}
      <span className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold">
        {infoMessage}
      </span>
      {iconRight && (
        <span className="flex items-center">
          <FontAwesomeIcon className="ml-1 text-current" icon={iconRight} size="sm" />
        </span>
      )}
    </div>
  );
};

export default Tag;
