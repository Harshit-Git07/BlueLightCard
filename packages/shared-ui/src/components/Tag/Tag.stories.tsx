import { Meta, StoryFn } from '@storybook/react';
import Tag from '.';
import { TagState } from './types';
import { faMinus, faPlus } from '@fortawesome/pro-regular-svg-icons';
import { faCircleBolt } from '@fortawesome/pro-solid-svg-icons';
import { library } from '@fortawesome/fontawesome-svg-core';
const icons = { faPlus, faMinus, faCircleBolt };
library.add(...Object.values(icons));

const iconArgSelect = {
  options: ['none', ...Object.keys(icons)],
  mapping: { none: undefined, ...icons },
  control: {
    type: 'select',
    labels: {
      faCircleBolt: 'Thunderbolt',
      faPlus: 'Plus Icon',
      faMinus: 'Minus Icon',
      none: 'No Icon',
    },
  },
};

const meta: Meta<typeof Tag> = {
  title: 'Component System/Tag',
  component: Tag,
  argTypes: {
    iconLeft: {
      description: 'Icon',
      ...iconArgSelect,
    },
    iconRight: {
      description: 'Icon',
      ...iconArgSelect,
    },
  },
};

const DefaultTemplate: StoryFn<typeof Tag> = (args) => <Tag {...args}></Tag>;

export const TagDefault = DefaultTemplate.bind({});
TagDefault.args = {
  state: TagState.Default,
  infoMessage: 'Tag',
  iconLeft: faCircleBolt,
  iconRight: faCircleBolt,
};

export const TagSuccess = DefaultTemplate.bind({});
TagSuccess.args = {
  state: TagState.Success,
  infoMessage: 'Tag',
  iconLeft: faCircleBolt,
  iconRight: faCircleBolt,
};

export const TagWarning = DefaultTemplate.bind({});
TagWarning.args = {
  state: TagState.Warning,
  infoMessage: 'Tag',
  iconLeft: faCircleBolt,
  iconRight: faCircleBolt,
};

export const TagError = DefaultTemplate.bind({});
TagError.args = {
  state: TagState.Error,
  infoMessage: 'Tag',
  iconLeft: faCircleBolt,
  iconRight: faCircleBolt,
};

export const TagInfo = DefaultTemplate.bind({});
TagInfo.args = {
  state: TagState.Info,
  infoMessage: 'Tag',
  iconLeft: faCircleBolt,
  iconRight: faCircleBolt,
};

export default meta;
