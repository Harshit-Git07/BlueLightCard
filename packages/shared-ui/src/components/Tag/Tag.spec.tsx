import Tag from './index';
import { TagState } from './types';
import renderer from 'react-test-renderer';
import { faCircleBolt } from '@fortawesome/pro-solid-svg-icons';

describe('Tag', () => {
  describe('snapshot Test', () => {
    it('renders a tag in default state', () => {
      const component = renderer.create(
        <Tag infoMessage="Tag" iconLeft={faCircleBolt} iconRight={faCircleBolt}></Tag>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a tag in Success state', () => {
      const component = renderer.create(
        <Tag
          infoMessage="Tag"
          iconLeft={faCircleBolt}
          iconRight={faCircleBolt}
          state={TagState.Success}
        ></Tag>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a tag in Error state', () => {
      const component = renderer.create(
        <Tag
          infoMessage="Tag"
          iconLeft={faCircleBolt}
          iconRight={faCircleBolt}
          state={TagState.Error}
        ></Tag>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a tag in Warning state', () => {
      const component = renderer.create(
        <Tag
          infoMessage="Tag"
          iconLeft={faCircleBolt}
          iconRight={faCircleBolt}
          state={TagState.Warning}
        ></Tag>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a tag in Info state', () => {
      const component = renderer.create(
        <Tag
          infoMessage="Tag"
          iconLeft={faCircleBolt}
          iconRight={faCircleBolt}
          state={TagState.Info}
        ></Tag>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a tag Without an Icon', () => {
      const component = renderer.create(<Tag infoMessage="Tag" state={TagState.Warning}></Tag>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a tag With 1 Icon', () => {
      const component = renderer.create(
        <Tag infoMessage="Tag" iconLeft={faCircleBolt} state={TagState.Warning}></Tag>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
