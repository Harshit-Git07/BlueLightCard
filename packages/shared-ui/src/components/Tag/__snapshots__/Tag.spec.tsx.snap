// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag snapshot Test renders a tag With 1 Icon 1`] = `
<div
  className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-warning-bright-light dark:bg-colour-warning-bright-dark text-colour-onWarning-bright-light dark:text-colour-onWarning-bright-dark border-[0.5px] border-colour-onWarning-bright-light dark:border-colour-onWarning-bright-dark rounded"
>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm mr-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
  <span
    className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
  >
    Tag
  </span>
</div>
`;

exports[`Tag snapshot Test renders a tag Without an Icon 1`] = `
<div
  className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-warning-bright-light dark:bg-colour-warning-bright-dark text-colour-onWarning-bright-light dark:text-colour-onWarning-bright-dark border-[0.5px] border-colour-onWarning-bright-light dark:border-colour-onWarning-bright-dark rounded"
>
  <span
    className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
  >
    Tag
  </span>
</div>
`;

exports[`Tag snapshot Test renders a tag in Error state 1`] = `
<div
  className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-error-bright-light dark:bg-colour-error-bright-dark text-colour-onError-bright-light dark:text-colour-onError-bright-dark border-[0.5px] border-colour-onError-bright-light dark:border-color-onError-bright-dark rounded"
>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm mr-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
  <span
    className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
  >
    Tag
  </span>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm ml-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
</div>
`;

exports[`Tag snapshot Test renders a tag in Info state 1`] = `
<div
  className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-info-bright-light dark:bg-colour-info-bright-dark text-colour-onInfo-bright-light dark:text-colour-onInfo-bright-dark border-[0.5px] border-colour-onInfo-bright-light dark:border-colour-onInfo-bright-dark rounded"
>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm mr-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
  <span
    className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
  >
    Tag
  </span>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm ml-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
</div>
`;

exports[`Tag snapshot Test renders a tag in Success state 1`] = `
<div
  className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-success-bright-light dark:bg-colour-success-bright-dark text-colour-onSuccess-bright-light dark:text-colour-onSuccess-bright-dark border-[0.5px] border-colour-onSuccess-bright-light dark:border-colour-onSuccess-bright-dark rounded"
>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm mr-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
  <span
    className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
  >
    Tag
  </span>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm ml-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
</div>
`;

exports[`Tag snapshot Test renders a tag in Warning state 1`] = `
<div
  className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-warning-bright-light dark:bg-colour-warning-bright-dark text-colour-onWarning-bright-light dark:text-colour-onWarning-bright-dark border-[0.5px] border-colour-onWarning-bright-light dark:border-colour-onWarning-bright-dark rounded"
>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm mr-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
  <span
    className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
  >
    Tag
  </span>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm ml-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
</div>
`;

exports[`Tag snapshot Test renders a tag in default state 1`] = `
<div
  className="inline-flex h-6 min-h-6 py-1 px-2 justify-center items-center gap-1 bg-colour-surface-container-light dark:bg-colour-surface-container-dark text-colour-onSurface-light dark:text-colour-onSurface-dark border-[0.5px] border-colour-onSurface-light dark:border-colour-onSurface-dark rounded"
>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm mr-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
  <span
    className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold"
  >
    Tag
  </span>
  <span
    className="flex items-center"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-bolt fa-sm ml-1 text-current"
      data-icon="circle-bolt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm74.1-396.4c5.8 4.7 7.6 12.9 4.2 19.6L281.9 240H352c6.8 0 12.9 4.3 15.1 10.7s.2 13.5-5.1 17.8l-160 128c-5.9 4.7-14.2 4.7-20.1-.1s-7.6-12.9-4.3-19.6L230.1 272H160c-6.8 0-12.8-4.3-15.1-10.7s-.2-13.5 5.1-17.8l160-128c5.9-4.7 14.2-4.7 20.1 .1z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </span>
</div>
`;
